snp_motion_planning/src/planning_server.cpp

```diff
@@ -369,10 +381,17 @@ class PlanningServer
 
         // Discrete contact check profile
         auto contact_check_lvs = get<double>(node_, LVS_PARAM);
-        auto contact_check_dist = get<double>(node_, CONTACT_DIST_PARAM);
-        profile_dict->addProfile<tesseract_planning::ContactCheckProfile>(
-            CONTACT_CHECK_DEFAULT_NAMESPACE, PROFILE,
-            std::make_shared<tesseract_planning::ContactCheckProfile>(contact_check_lvs, contact_check_dist));
+        auto contact_check_profile =
+            std::make_shared<tesseract_planning::ContactCheckProfile>(contact_check_lvs, min_contact_dist);
+        contact_check_profile->config.contact_request.type = tesseract_collision::ContactTestType::FIRST;
+        contact_check_profile->config.contact_manager_config.margin_data_override_type =
+            tesseract_common::CollisionMarginOverrideType::MODIFY;
+        for (const UniqueCollisionPair& pair : collision_pairs)
+          contact_check_profile->config.contact_manager_config.margin_data.setPairCollisionMargin(
+              pair.first, pair.second, pair.distance);
+
+        profile_dict->addProfile<tesseract_planning::ContactCheckProfile>(CONTACT_CHECK_DEFAULT_NAMESPACE, PROFILE,
```

Comment:
I feel like creating the `ContactCheckProfile` should be a method in the `planner_profiles.hpp` file.

Reply:
I thought about that too; I'll move it